server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java

```diff
@@ -534,6 +496,27 @@ private void createIndex(SystemIndexMigrationInfo migrationInfo, ActionListener<
         );
     }
 
+    private void createIndexRetryOnFailure(SystemIndexMigrationInfo migrationInfo, ActionListener<ShardsAcknowledgedResponse> listener) {
+        createIndex(migrationInfo, ActionListener.wrap(listener::onResponse, e -> {
+            logger.warn("createIndex failed, retrying after removing index [{}] from previous attempt", migrationInfo.getNextIndexName());
+            deleteIndex(migrationInfo, ActionListener.wrap(cleanupResponse -> createIndex(migrationInfo, listener), e2 -> {
+                logger.warn("createIndex failed after retrying, aborting", e2);
+                listener.onFailure(e2);
+            }));
+        }));
+    }
+
+    private <T> void deleteIndex(SystemIndexMigrationInfo migrationInfo, ActionListener<AcknowledgedResponse> listener) {
+        logger.info("removing index [{}] from feature [{}]", migrationInfo.getNextIndexName(), migrationInfo.getFeatureName());
+        String newIndexName = migrationInfo.getNextIndexName();
+        baseClient.admin().indices().prepareDelete(newIndexName).execute(ActionListener.wrap(ackedResponse -> {
```

Comment:
This code is identical to the prior `cleanUpPreviousMigration` code except for that I'm using a `baseClient` here since I don't have access to the `migrationInfo` needed for `migrationInfo.createClient(baseClient)`.  This previous call produced an Origin Setting Client.  I don't think we need that here, though I'm not 100% on this.